### PR TITLE
CVE-2016-1690 and CVE-2013-2842	

### PR DESCRIPTION
**Pull request by Pearse4004**
Edited yml files for two UAF vulnerabilities that were fixed in chromium.

### CODE REVIEW

File: cves/CVE-2016-1690.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: A user could set an input value that could cause the form being viewed to be deleted. This would then cause the data from the autofil to be deleted as well. After all this data has been deleted there could still be process running and trying to use the deleted data from the auto fill which results in a crash.
```
Comment (Nitepone) [inline cves/CVE-2016-1690.yml]: High character count on this line, might want to use a pipe to have a multiline description

File: cves/CVE-2016-1690.yml [hunk 2]
```diff
@@ -127,12 +127,12 @@ interesting_commits:
...
 
     If there are no interesting commits, demonstrate that you completed this section by explaining what happened between the VCCs and the fix.
-  answer:
+  answer: There was one interesting commit, there was a commit to potentially fix the autofill crash. It was similar to the vcc fix where it checks to see if the element is not null.
   commits:
-  - commit: 
-    note: 
-  - commit: 
-    note: 
+  - commit: d80f2526c38b3575f78d1e3335b2423110438c37
```
Comment (Nitepone) [inline cves/CVE-2016-1690.yml]: This commit looks to be the bug fix, I would add a note to make that clear

File: cves/CVE-2013-2842.yml [hunk 3]
```diff
@@ -103,20 +105,20 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave the entries blank except for "answer". Write down where you looked in "answer".
-  answer: No evidence on how this was found.
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: The bug report for the cve
```
Comment (CatherineOsadciw) [inline cves/CVE-2013-2842.yml]: How was the vulnerability discovered to result in the report being filed?

File: cves/CVE-2013-2842.yml [hunk 4]
```diff
@@ -167,40 +168,40 @@ lessons:
...
+    applies:
+    note:
   native_wrappers:
-    applies: 
-    note: 
+    applies:
+    note:
   distrust_input:
     applies: true
     note: While there was not really input to the system, it could be attacked by
```
Comment (CatherineOsadciw) [inline cves/CVE-2013-2842.yml]: What parts are you referring to?

File: cves/CVE-2013-2842.yml [hunk 5]
```diff
@@ -213,7 +214,6 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: "The mistake was that they freed a portion of memory that went along with
-    a frame in the renderer, but they did not remove all references to it.  \nThe
-    fix was to check the frame before it was used to make sure that it had not been
-    freed yet.\n"
+  answer: The mistake was that they freed the memory of the rendered areana but
```
Comment (CatherineOsadciw) [inline cves/CVE-2013-2842.yml]: Sounds like this was a coding mistake.  Remove third person references (i.e. uses of "they") to sound more professional.

File: cves/CVE-2016-1690.yml [hunk 6]
```diff
@@ -32,7 +32,11 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: A user could set an input value that could cause the form being
```
Comment (CatherineOsadciw) [inline cves/CVE-2016-1690.yml]: Is this referring to any form inputs on a webpage?

File: cves/CVE-2016-1690.yml [hunk 7]
```diff
@@ -86,9 +90,9 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: Yes
```
Comment (CatherineOsadciw) [inline cves/CVE-2016-1690.yml]: Need more detail than just "yes" -- what kind of tests did the code originally include? Did they miss a specific case that would have covered this vulnerability?

File: cves/CVE-2016-1690.yml [hunk 8]
```diff
@@ -104,20 +108,20 @@ discovered:
...
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: The bug report for the cve
+  date: 2016-04-29
+  automated: false
+  google: false
+  contest: rob@robwu.nl
```
Comment (CatherineOsadciw) [inline cves/CVE-2016-1690.yml]: Contest should be "true" or "false." You can put this information in the "answer" section as additional info.

File: cves/CVE-2016-1690.yml [hunk 9]
```diff
@@ -127,12 +131,15 @@ interesting_commits:
     emerging themes?
 
     If there are no interesting commits, demonstrate that you completed this section by explaining what happened between the VCCs and the fix.
-  answer:
+  answer: There was one interesting commit, there was a commit to potentially
```
Comment (CatherineOsadciw) [inline cves/CVE-2016-1690.yml]: So was this commit not the fix?  Did it not work initially?

File: cves/CVE-2016-1690.yml [hunk 10]
```diff
@@ -163,38 +170,39 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies: 
-    note: 
+    applies: true
+    note: the program could still run once they check that the form was deleted
```
Comment (CatherineOsadciw) [inline cves/CVE-2016-1690.yml]: Sounds like this is more like "distrust input"

File: cves/CVE-2016-1690.yml [hunk 11]
```diff
@@ -207,4 +215,5 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: The design mistake was made by not checking input and trusting that
```
Comment (CatherineOsadciw) [inline cves/CVE-2016-1690.yml]: Add some more detail to your analysis here

File: cves/CVE-2016-1690.yml (hunk 9 above)
Comment (Pearse4004) [reply cves/CVE-2016-1690.yml]: Correct it doesn't look like the commit was ran against a test

File: cves/CVE-2016-1690.yml (hunk 6 above)
Comment (Pearse4004) [reply cves/CVE-2016-1690.yml]: Any text input that a user can autofill

File: cves/CVE-2013-2842.yml (hunk 3 above)
Comment (Pearse4004) [reply cves/CVE-2013-2842.yml]: No indication of how it was found

File: cves/CVE-2013-2842.yml (hunk 5 above)
Comment (Pearse4004) [reply cves/CVE-2013-2842.yml]: Fixed

**Pull request merged**